python/rpdk/java/templates/init/guided_aws/StubCreateHandler.java

```diff
@@ -29,157 +26,92 @@
 
         this.logger = logger;
 
-        final ResourceModel model = request.getDesiredResourceState();
-
         // TODO: Adjust Progress Chain according to your implementation
         // https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/proxy/CallChain.java
 
-        return ProgressEvent.progress(model, callbackContext)
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
 
             // STEP 1 [check if resource already exists]
             // if target API does not support 'ResourceAlreadyExistsException' then following check is required
             // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
-            //.then(progress -> checkForPreCreateResourceExistence(proxy, request, progress))
+            .then(progress ->
+                // STEP 1.0 [initialize a proxy context]
+                // If your service API is not idempotent, meaning it does not distinguish duplicate create requests against some identifier (e.g; resource Name)
+                // and instead returns a 200 even though a resource already exists, you must first check if the resource exists here
+                // NOTE: If your service API throws 'ResourceAlreadyExistsException' for create requests this method is not necessary
+                proxy.initiate("{{ call_graph }}::{{ operation }}::PreExistanceCheck", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+
+                    // STEP 1.1 [TODO: construct a body of a request]
+                    .translateToServiceRequest(Translator::translateToReadRequest)
+
+                    // STEP 1.2 [TODO: make an api call]
+                    .makeServiceCall((awsRequest, client) -> {
+                        AwsResponse awsResponse = null;
+
+                        // TODO: add custom read resource logic
+
+                        logger.log(String.format("%s has successfully been read.", ResourceModel.TYPE_NAME));
+                        return awsResponse;
+                    })
```

Comment:
```suggestion
                    .makeServiceCall((awsRequest, sdkProxyClient) -> sdkProxyClient.injectCredentialsAndInvokeV2(awsRequest, sdkProxyClient.client()::describeResource))
```

Reply:
not sure this will build bc sdkProxyClient isnt defined in mocked example, so there is no `describeResource` method